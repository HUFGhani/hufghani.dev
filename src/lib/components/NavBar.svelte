<script lang="ts">
	import Random from '$lib/icon/Random.svelte'
	import { AUTHORNAME, MENULINK } from '$lib/info'

	const selectRandomTheme = (): void => {
		let randomNumber = Math.floor(Math.random() * 29) + 1

		const themes = (themeId: number) =>
			({
				1: 'light',
				2: 'dark',
				3: 'cupcake',
				4: 'bumblebee',
				5: 'emerald',
				6: 'corporate',
				7: 'synthwave',
				8: 'retro',
				9: 'cyberpunk',
				10: 'valentine',
				11: 'halloween',
				12: 'garden',
				13: 'forest',
				14: 'aqua',
				15: 'lofi',
				16: 'pastel',
				17: 'fantasy',
				18: 'wireframe',
				19: 'black',
				20: 'luxury',
				21: 'dracula',
				22: 'cmyk',
				23: 'autumn',
				24: 'business',
				25: 'acid',
				26: 'lemonade',
				27: 'night',
				28: 'coffee',
				29: 'winter',
			}[themeId])

		const theme = localStorage.getItem('theme')
		if (theme != undefined && theme != '') {
			if (localStorage.getItem('theme') && localStorage.getItem('theme') != '') {
				document.documentElement.setAttribute('data-theme', theme)
			}
		}
		document.documentElement.setAttribute('data-theme', themes(randomNumber)?.toString()!)
		localStorage.setItem('theme', document.documentElement.getAttribute('data-theme')!)
	}
</script>

<div
	class="flex items-center 
	sm:flex-col md:flex-row lg:flex-row xl:flex-row 
	justify-center lg:justify-between xl:justify-between 
	flex-wrap w-full pt-10 
	sm:px-4 md:px-4 lg:px-8 xl:px-8 
	bg-transparent"
>
	<div>
		<a href="/">
			<h1 class="logo font-normal text-4xl">
				{AUTHORNAME}
			</h1>
		</a>
	</div>

	<div class="flex flex-row md:pl-10 space-x-4 items-center justify-center text-base text-center">
		{#each MENULINK as { link, title, isExternal, abbr, fullTitle }}
			<div class="rounded-md p-2 text-center hover:bg-accent">
				{#if isExternal}
					<a href={link} aria-label={title} target="_blank">
						<abbr title={fullTitle} class="border-none cursor-default no-underline">
							{title}
						</abbr>
					</a>
				{:else}
					<a href={link} aria-label={title}>{title} </a>
				{/if}
			</div>
		{/each}
		<button
			class="rounded-md p-2 text-center hover:bg-accent"
			aria-label="random theme selector"
			on:click={selectRandomTheme}
		>
			<Random cssClasses="w-6 h-6" />
		</button>
	</div>
</div>

<style>
	.logo {
		font-family: Dr Sugiyama, cursive;
		background-image: linear-gradient(60deg, #7928ca, #ffb03a);
		background-clip: text;
		color: transparent;
	}
</style>
