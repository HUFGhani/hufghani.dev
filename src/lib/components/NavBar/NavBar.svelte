<script lang="ts">
	import { Logo } from '$lib/components/Logo';
	import { MENULINK } from '$lib/config';
	import { Random } from '$lib/icon';
	import { getTheme, setHTMLAttribute, setTheme } from '$lib/utlis';


	const selectRandomTheme = (): void => {
		let randomNumber = Math.floor(Math.random() * 29) + 1;
		const themes = (themeId: number) =>
			({
				1: 'light',
				2: 'dark',
				3: 'cupcake',
				4: 'bumblebee',
				5: 'emerald',
				6: 'corporate',
				7: 'synthwave',
				8: 'retro',
				9: 'cyberpunk',
				10: 'valentine',
				11: 'halloween',
				12: 'garden',
				13: 'forest',
				14: 'aqua',
				15: 'lofi',
				16: 'pastel',
				17: 'fantasy',
				18: 'wireframe',
				19: 'black',
				20: 'luxury',
				21: 'dracula',
				22: 'cmyk',
				23: 'autumn',
				24: 'business',
				25: 'acid',
				26: 'lemonade',
				27: 'night',
				28: 'coffee',
				29: 'winter'
			})[themeId];

		const theme = getTheme();
		if (theme != undefined && theme != '') {
			if (localStorage.getItem('theme') && getTheme() != '') {
				setHTMLAttribute(theme);
			}
		}

		setTheme(themes(randomNumber)?.toString()!);
	};
</script>

<div
	class="flex w-full
	flex-wrap items-center justify-center bg-transparent
	pb-4 align-middle sm:flex-col
	sm:px-4 md:flex-row
	md:px-4 lg:flex-row lg:justify-between lg:px-8
	xl:flex-row xl:justify-between
	 xl:px-8"
>
	<div class="mb-4 lg:mb-0 xl:mb-0">
		<a href="/" aria-label="Go to Homepage">
			<Logo />
		</a>
	</div>

	<div class="flex flex-row items-center justify-center space-x-4 text-center text-base md:pl-10">
		{#each MENULINK as { link, title, isExternal, fullTitle }}
			{#if title !== 'Blog'}
				<div data-sveltekit-preload-data="hover" class="hover:bg-accent rounded-md p-2 text-center">
					{#if isExternal}
						<a href={link} aria-label={title} target="_blank" rel="noreferrer" class="h-12 w-12">
							<abbr title={fullTitle} class="cursor-default border-none no-underline">
								{title}
							</abbr>
						</a>
					{:else}
						<a class="h-12 w-12" href={link} aria-label={title}>{title} </a>
					{/if}
				</div>
			{/if}
		{/each}
		<button
			class="hover:bg-accent rounded-md p-2 text-center"
			aria-label="random theme selector"
			on:click={selectRandomTheme}
		>
			<Random cssClasses="w-6 h-6" />
		</button>
	</div>
</div>
