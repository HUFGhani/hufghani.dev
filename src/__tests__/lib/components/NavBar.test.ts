import NavBar from '$lib/components/NavBar.svelte'
import { render } from '@testing-library/svelte'

describe(`NavBar/>`, () => {
	it(`should render NavBar`, () => {
		const { container } = render(NavBar)
		expect(container).toMatchSnapshot()
	})
})
