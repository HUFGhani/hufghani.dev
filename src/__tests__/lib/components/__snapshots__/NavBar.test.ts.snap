// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> should render NavBar 1`] = `
<body>
  <div>
    <div
      class="flex items-center sm:flex-col md:flex-row lg:flex-row xl:flex-row justify-center lg:justify-between xl:justify-between flex-wrap w-full sm:px-4 md:px-4 lg:px-8 xl:px-8 bg-transparent pb-4 align-middle"
    >
      <div>
        <a
          href="/"
        >
          <h1
            class="logo font-normal text-4xl svelte-3g66w"
          >
            Hamza U. F. Ghani
          </h1>
        </a>
      </div>
       
      <div
        class="flex flex-row md:pl-10 space-x-4 items-center justify-center text-base text-center"
      >
        <div
          class="rounded-md p-2 text-center hover:bg-accent"
        >
          <a
            aria-label="Project"
            href="project"
            sveltekit:prefetch=""
          >
            Project
          </a>
        </div>
        <div
          class="rounded-md p-2 text-center hover:bg-accent"
        >
          <a
            aria-label="Blog"
            href="blog"
            sveltekit:prefetch=""
          >
            Blog
          </a>
        </div>
        <div
          class="rounded-md p-2 text-center hover:bg-accent"
        >
          <a
            aria-label="CV"
            href="docs/Curriculum-Vitae.pdf"
            target="_blank"
          >
            <abbr
              class="border-none cursor-default no-underline"
              title="Curriculum Vitae"
            >
              CV
            </abbr>
          </a>
        </div>
         
        <button
          aria-label="random theme selector"
          class="rounded-md p-2 text-center hover:bg-accent"
        >
          <svg
            class="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M424.1 287c-15.13-15.12-40.1-4.426-40.1 16.97V352H336L153.6 108.8C147.6 100.8 138.1 96 128 96H32C14.31 96 0 110.3 0 128s14.31 32 32 32h80l182.4 243.2C300.4 411.3 309.9 416 320 416h63.97v47.94c0 21.39 25.86 32.12 40.99 17l79.1-79.98c9.387-9.387 9.387-24.59 0-33.97L424.1 287zM336 160h47.97v48.03c0 21.39 25.87 32.09 40.1 16.97l79.1-79.98c9.387-9.391 9.385-24.59-.0013-33.97l-79.1-79.98c-15.13-15.12-40.99-4.391-40.99 17V96H320c-10.06 0-19.56 4.75-25.59 12.81L254 162.7L293.1 216L336 160zM112 352H32c-17.69 0-32 14.31-32 32s14.31 32 32 32h96c10.06 0 19.56-4.75 25.59-12.81l40.4-53.87L154 296L112 352z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
